src/controllers/flamegraphController.ts

```diff
@@ -0,0 +1,78 @@
+//middelware function that will take a .perf and return a .svg
+/*
+1. run the following command to collapse the stack frames
+./stackcollapse-perf.pl out.perf > out.folded
+
+2. run the following command to convert collapsed stack frames into .svg
+./flamegraph.pl out.kern_folded > kernel.svg
+
+3. return the .svg to the next middleware function
+*/
+
+// exec() method: This method creates a shell first and then executes the command. 
+const exec = require('child_process').exec   
+import { Express, Request, Response, NextFunction } from 'express'
+import * as stream from 'stream'
+
+const express = require('express');
+const app = express();
+
+type FlamegraphSVGController = {
+  //collapse the stack traces in the perf:
+  stackCollapse: (req: Request, res: Response, next: NextFunction) => void, 
+  //convert the collapsed stack traces into an SVG:
+  toSVG: (req: Request, res: Response, next: NextFunction) => void
+}
+
+const flamegraphController: FlamegraphSVGController = {
+  stackCollapse: (req: Request, res: Response, next: NextFunction): void => {
+    //node child process
+    const fileName = res.locals.id; 
+    console.log(fileName, '-----')
+    exec(`./src/perlScripts/stackCollapse-perf.pl ./database/captures/${fileName}.perf > ./database/folded/${fileName}.folded`,
+       (error: stream.Readable, stdout: stream.Readable, stderr: stream.Readable) => {
+        //change to String | Buffer if it doesnt work
+        if (error) {
+            return next({
+              log: 'something went wrong with the stackFolder middleware',
+              message: { err: error },
+            });
+          // console.log('err in stackCollapse');
+          // return res.status(500).json('stackFolder err-----');
+        }
+        return next();
+      })
+  },
+
+  toSVG: (req: Request, res: Response, next: NextFunction): void => {
+    const fileName = res.locals.id;
+    console.log(fileName, '-----')
+    exec(`./src/perlScripts/flamegraph.pl ./database/folded/${fileName}.folded > ./database/SVGs/${fileName}.svg`,
+      (error: Error, stdout: string | Buffer, stderr: string | Buffer) => {
+      //store file ./database/SVGs
+      if (!error) return next();
+      else 
+      // return res.status(500).json('toSVG err-----');
+      return next({
+        log: 'something went wrong with the toSVG middleware',
```

Comment:
Our [InfernodeError](https://github.com/oslabs-beta/infernode/blob/2017056f343d798420b812415f4616191148de06/src/utils/globalErrorHandler.ts#L4) type does not have a `log` property, but rather `message` (for internal logs) and `userMessage` (for user facing error messages) properties. `message` is inherited from the extended `Error` class, do we need an additional `log` property to cover all use cases? The implementation currently logs all fields on an `InfernodeError` object.

Reply:
For now, we pass error.message to the userMessage to match a string type. But can we change the userMessage type from string to an Error object so that we can pass all info from the error instance generated from the async function?